File: js/3300.0_Minimum_Element_After_Replacement_With_Digit_Sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minElement = function(nums) {
    let ans = Infinity;
    for (let x of nums) {
        let s = 0;
        while (x > 0) {
            s += x % 10;
            x = Math.floor(x / 10);
        }
        ans = Math.min(ans, s);
    }
    return ans;
};


/**
 * @param {number[]} nums
 * @return {number}
 */
var minElement = function(nums) {
    let ans = Infinity;
    for (let x of nums) {
        let s = 0;
        while (x > 0) {
            s += x % 10;
            if (s > ans) break;
            x = Math.floor(x / 10);
        }
        ans = Math.min(ans, s);
    }
    return ans;
};

File: leetcode-cn/3300.0_Minimum_Element_After_Replacement_With_Digit_Sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        def digitSum(x: int) -> int:
            s = 0
            while x:
                s += x % 10
                x //= 10
            return s

        return min(digitSum(x) for x in nums)


class Solution:
    def minElement(self, nums: List[int]) -> int:
        def digitSum(x: int) -> int:
            s = 0
            while x:
                x, r = divmod(x, 10)
                s += r
            return s

        return min(digitSum(x) for x in nums)


class Solution:
    def minElement(self, nums: List[int]) -> int:
        ans = inf
        for x in nums:
            s = 0
            while x:
                s += x % 10
                if s >= ans:
                    break
                x //= 10
            ans = min(ans, s)
        return ans